tests/test_smoothquant/test_llama_attention.py

```diff
@@ -55,22 +54,34 @@ def torch_context_attention(xq, xk, xv, bs, seqlen, num_head, head_dim):
     reason="triton requires cuda version to be higher than 11.4 or not install torch_int",
 )
 def test_llama_context_attention():
-    head_num = 8
+    head_num = 2
     seq_len = 32
     head_dim = 64
     dtype = torch.float
     hidden_size = head_num * head_dim
 
     smooth_attn = LLamaSmoothquantAttention(head_num * head_dim, head_num)
 
-    smooth_attn.q_proj.weight = torch.ones(hidden_size, hidden_size).to(torch.int8)
-    smooth_attn.k_proj.weight = torch.ones(hidden_size, hidden_size).to(torch.int8)
-    smooth_attn.v_proj.weight = torch.ones(hidden_size, hidden_size).to(torch.int8)
-    smooth_attn.out_proj.weight = torch.ones(hidden_size, hidden_size).to(torch.int8)
+    smooth_attn.q_proj.weight = torch.ones(hidden_size, hidden_size, device="cuda").to(torch.int8)
+    smooth_attn.k_proj.weight = torch.ones(hidden_size, hidden_size, device="cuda").to(torch.int8)
+    smooth_attn.v_proj.weight = torch.ones(hidden_size, hidden_size, device="cuda").to(torch.int8)
+    smooth_attn.out_proj.weight = torch.ones(hidden_size, hidden_size, device="cuda").to(torch.int8)
+    smooth_attn.out_proj.weight[:, 1:hidden_size] = torch.zeros(hidden_size - 1, device="cuda").to(torch.int8)
+
+    qkv_weight_scale = 1.0
+
+    ones = torch.ones(hidden_size, hidden_size, dtype=torch.float, device="cuda")
 
     smooth_attn = smooth_attn.to("cuda")
 
-    input = torch.randint(-127, 127, (1, seq_len, head_num * head_dim), dtype=torch.int8, device="cuda")
+    input = torch.randint(-20, 20, (1, seq_len, head_num * head_dim), dtype=torch.int8, device="cuda")
+    input_scale = 1 / 20.0
+
+    output = torch.matmul(input.to(torch.float) * input_scale, ones)
+    qkv_max_out = torch.max(torch.abs(output)) / 127
```

Comment:
Just wonder: why 127

Reply:
The range of int8 is [-128, 127]. When value divide the qkv_max_out, the ouput value can be in [-128, 127]